src/scheduler/simulate.c

```diff
@@ -1453,11 +1446,6 @@ add_event(event_list *calendar, timed_event *te)
 
 	calendar->events = add_timed_event(calendar->events, te);
 
-	if (te->event_type & TIMED_RUN_EVENT)
-		((resource_resv *)te->event_ptr)->run_event = te;
-	else if (te->event_type & TIMED_END_EVENT)
-		((resource_resv *)te->event_ptr)->end_event = te;
```

Comment:
don't you need to modify add_timed_event() to set run_event and end_event pointers?

Reply:
We don't need to set the pointers when the event is added since they are set in create_event().  The idea behind this change was to set the pointers in one place and NULL them in one place that everybody calls.  All events are created via create_event() and freed via free_timed_event().